Makefile

```
@@ -226,20 +228,27 @@ POLKADOT_VERSION=$(shell awk -F "=" '/name = "polkadot-cli"/,/version = ".*"/{ p
 .PHONY: version
 version:
 ifndef v
-	@echo "Please set the version with v=X.X.X-X"
+	@echo "Please set the version with v=#.#.#[-#]"
 	@exit 1
 endif
 ifneq (,$(findstring v,  $(v)))
-	@echo "Please don't prefix with a 'v'. Use: v=X.X.X-X"
+	@echo "Please don't prefix with a 'v'. Use: v=#.#.#[-#]"
 	@exit 1
 endif
 ifeq (,$(POLKADOT_VERSION))
 	@echo "Error: Having trouble finding the Polkadot version. Sorry about that.\nCheck my POLKADOT_VERSION variable command."
 	@exit 1
 endif
 	@echo "Setting the crate versions to "$(v)+polkadot$(POLKADOT_VERSION)
-	find ./ -type f -name 'Cargo.toml' -exec sed -i '' 's/^version = \"0\.0\.0\"/version = \"$(v)+polkadot$(POLKADOT_VERSION)\"/g' {} \;
-	$(MAKE) check
+ifeq ($(UNAME), Linux)
+	$(eval $@_SED := -i -e)
+endif
+ifeq ($(UNAME), Darwin)
+	$(eval $@_SED := -i '')
+endif
+	find . -type f -name "Cargo.toml" -print0 | xargs -0 sed $($@_SED) 's/^version = \"0\.0\.0\"/version = \"$(v)+polkadot$(POLKADOT_VERSION)\"/g';
+	@echo "Doing cargo check with fake examples project seems to be the easiest way to update version in Cargo.lock"
```

These are doc examples we are checking not a fake project. There just are very few doc examples (currently none).
```suggestion
	@echo "Doing cargo check for just examples seems to be the easiest way to update version in Cargo.lock"
```